File: UniShare.API/Models/DTOs/RentalRequests/RentalRequestSummaryDto.cs
namespace UniShare.API.Models.DTOs.RentalRequests;

public class RentalRequestSummaryDto
{
    public Guid Id { get; set; }
    public string Status { get; set; } = null!;
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public decimal TotalPrice { get; set; }
    public decimal? DepositAmount { get; set; }
    public DateTime CreatedAt { get; set; }

    // Listing info
    public Guid ListingId { get; set; }
    public string ListingTitle { get; set; } = null!;
    public string? ListingImageUrl { get; set; }

    // Counterpart (the other participant from the viewer's perspective)
    public Guid OtherParticipantId { get; set; }
    public string OtherParticipantName { get; set; } = null!;
    public string? OtherParticipantAvatarUrl { get; set; }

    /// <summary>"requester" or "owner" — the viewer's role in this request.</summary>
    public string Role { get; set; } = null!;
}

File: UniShare.APP/lib/features/rentals/models/rental_request_summary_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'rental_request_summary_dto.g.dart';

/// Rental request summary matching backend RentalRequestSummaryDto.
///
/// Flat fields used in list endpoints. The `role` field indicates
/// whether the current user is the "requester" or "owner".
@JsonSerializable()
class RentalRequestSummaryDto {
  final String id;
  final String status;
  final DateTime startDate;
  final DateTime endDate;
  final double totalPrice;
  final double? depositAmount;
  final DateTime createdAt;
  final String listingId;
  final String listingTitle;
  final String? listingImageUrl;
  final String otherParticipantId;
  final String otherParticipantName;
  final String? otherParticipantAvatarUrl;
  final String role;

  const RentalRequestSummaryDto({
    required this.id,
    required this.status,
    required this.startDate,
    required this.endDate,
    required this.totalPrice,
    this.depositAmount,
    required this.createdAt,
    required this.listingId,
    required this.listingTitle,
    this.listingImageUrl,
    required this.otherParticipantId,
    required this.otherParticipantName,
    this.otherParticipantAvatarUrl,
    required this.role,
  });

  factory RentalRequestSummaryDto.fromJson(Map<String, dynamic> json) =>
      _$RentalRequestSummaryDtoFromJson(json);

  Map<String, dynamic> toJson() => _$RentalRequestSummaryDtoToJson(this);
}
